src/zkscript/util/utility_scripts.py

```diff
@@ -109,6 +109,8 @@ def roll(position: int, n_elements: int) -> Script:
 
     if (position, n_elements) in patterns_to_roll:
         out += Script(patterns_to_roll[(position, n_elements)])
+    elif position == n_elements - 1:
```

Comment:
Agreed on the second comment (moving it at the beginning of the function).

If `position < n_elements - 1` then the function (and the pick function) should raise an error (we are trying to roll/pick more elements than the ones available on top of the stack).

There is no need for this case in the pick case because the code already handles that case. The case `position == n_elements-1` is special for `roll` because in that case we should do nothing (but we still have to duplicate elements if we are calling `pick`)